src/Http/AccessToken.php

```diff
@@ -95,7 +95,12 @@ public static function generate($userId)
      */
     public function touch(ServerRequestInterface $request = null)
     {
-        $this->last_activity_at = Carbon::now();
+        $now = Carbon::now();
+
+        // Only update if the current timestamp >120s ago
+        if ($this->last_activity_at === null || $this->last_activity_at->diffInSeconds($now) > 120) {
```

Comment:
Minor nitpick, but can we set it to >=, or maybe even the diff / 2? It would be preferable to keep this somewhat overlapping.